### PR TITLE
feat: add support for includeFiles

### LINKED ISSUE #6: feature request: include library root folder changes as affected
Both in NX and Traf right now, if I change for example the `package.json` of a library (which can affect its build) - it doesn't end up in the "affected" list, hence I need to manually change something silly in the code (a comment, ordering) in order to trigger a build.

Can we somehow add that these kind of changes do count as "affected"? it doesn't need to add into the affected results the libraries that depend on my change, but only the library itself

### PR DESCRIPTION
**Pull request by raz-ezra-lmnd**
Supporting an `includeFiles` flag to indicate non ts/js files that should trigger `affected` when changed

fixes #6

### CODE REVIEW

File: libs/core/src/true-affected.ts [hunk 1]
```diff
@@ -63,12 +65,19 @@ export const trueAffected = async ({
...
 
   const changedFiles = getChangedFiles({ base, cwd }).filter(
-    ({ filePath }) => project.getSourceFile(resolve(cwd, filePath)) != null
+    ({ filePath }) =>
+      includeFiles.some((fileName) => {
+        if (filePath.endsWith(fileName)) {
+          project.addSourceFileAtPath(resolve(cwd, filePath));
+          return true;
+        }
+        return false;
```
Comment (EladBezalel) [inline libs/core/src/true-affected.ts:79]: Why not just do it before? why in the filter?

File: libs/nx/src/cli.ts [hunk 2]
```diff
@@ -98,13 +101,17 @@ const affectedCommand: CommandModule<unknown, AffectedOptions> = {
       desc: 'Output affected projects as JSON',
       default: false,
     },
+    includeFiles: {
+      desc: 'Comma separated list of files to include',
```
Comment (EladBezalel) [inline libs/nx/src/cli.ts:105]:
```suggestion
      desc: 'Comma separated list of files to include',
      type: 'array',
```

File: libs/nx/src/cli.ts [hunk 3]
```diff
@@ -21,13 +21,15 @@ const affectedAction = async ({
...
   tsConfigFilePath,
+  includeFiles,
 }: AffectedOptions) => {
   const projects = await getNxTrueAffectedProjects(cwd);
   const affected = await trueAffected({
     cwd,
     rootTsConfig: tsConfigFilePath,
     base,
     projects,
+    includeFiles: includeFiles?.split(','),
```
Comment (EladBezalel) [inline libs/nx/src/cli.ts:32]: should come from yargs `type: 'array',`

File: libs/core/src/true-affected.ts [hunk 4]
```diff
@@ -63,8 +65,11 @@ export const trueAffected = async ({
     new Map<string, string[]>()
   );
 
-  projects.forEach(({ tsConfig }) => {
+  projects.forEach(({ tsConfig, sourceRoot }) => {
     project.addSourceFilesFromTsConfig(resolve(cwd, tsConfig));
+    includeFiles.forEach((path) => {
+      project.addSourceFilesAtPaths(`${resolve(cwd, sourceRoot)}/${path}`);
+    });
```
Comment (EladBezalel) [inline libs/core/src/true-affected.ts:72]:
```suggestion
    project.addSourceFilesAtPaths(includeFiles.map((path) => `${resolve(cwd, sourceRoot)}/${path}`))
```

File: libs/nx/src/cli.spec.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+jest.mock('node:child_process', () => ({
+  spawn: (command: string, options: Record<string, unknown>) =>
+    mockSpawn(command, options),
+}));
+
+async function runCommand(args: string[]) {
+  process.argv = ['node', 'cli.js', ...args];
+
+  // const { run } = await import('./cli');
```
Comment (EladBezalel) [inline libs/nx/src/cli.spec.ts:31]: ?

Comment (raz-ezra-lmnd) [reply libs/nx/src/cli.spec.ts]: Woops

**Review by EladBezalel — COMMENTED** [review]
1 last thing, please revert package lock 🙏

**Review by EladBezalel — APPROVED** [review]

**Pull request merged**